### DISCUSSION TITLE
Is there a cap on sub-trajectories?

### DISCUSSION THREAD

**Discussion by chrisreading01** [r-simmer/simmer#261] [Q&A] [answered]
I'm beginning to build some more complex branching trajectories, to simulate patient pathways from referral to discharge, through a number of branching activities. I've been using branching sub-trajectories to try and achieve this, but whenever I try to have a trajectory which goes reaches 6 layers of nested sub-trajectories (which is probably a bad idea but I am new to this!) my trajectories gets truncated. 

Is there a cap on the number of subtrajectories which is causing this? Below is a really rubbish but working example :) 

Thank you!

```
testTrajectory <- trajectory() %>% 
  branch(
    function(){return(1)}
    ,continue = FALSE
    ,trajectory() %>%
      branch(
        function(){return(1)}
        ,continue = FALSE
        ,trajectory() %>%
          branch(
            function(){return(1)}
            ,continue = FALSE
            ,trajectory() %>%
              branch(
                function(){return(1)}
                ,continue = FALSE
                ,trajectory() %>%
                  branch(
                    function(){return(1)}
                    ,continue = FALSE
                    ,trajectory() %>%
                      branch(
                        function(){return(1)}
                        ,continue = FALSE
                        ,trajectory() %>%
                          branch(
                            function(){return(1)}
                            ,continue = FALSE
                            ,trajectory() %>%
                              branch(
                                function(){return(1)}
                                ,continue = FALSE
                                ,trajectory() %>%
                                  timeout(1)
                              )
                          )
                      )
                  )
              )
          )
      )
  )

testTrajectory 

```

**Comment by Enchufa2** [comment, marked as answer]
No, there is no cap on the number of nested subtrajectories you can use. There **is**, however, a cap on the number of nested subtrajectories that are **printed** out. As you figured out, [the current limit is 6](https://github.com/r-simmer/simmer/blob/v4.4.3/inst/include/simmer/activity/fork.h#L56). I suppose that, at the time I wrote that, I thought no one would use so many sublevels, and anyway the output would become unreadable. But then, we have `simmer.plot`, which parses the output to plot diagrams of trajectories. So probably we should get rid of that limit.

**Reply by Enchufa2** [reply]
FYI, tracked in #262

**Reply by chrisreading01** [reply]
That makes a lot of sense! I did try putting it through simmer plot and saw the trajectories ending at 6 sublevels still so I didn't realise it was just a printing cap. Thank you for the clarification, and for such a great package!
